mod/ping.php

```diff
@@ -195,13 +196,20 @@ function ping_init(App $a)
 			}
 		}
 
-		$ev = qu("SELECT count(`event`.`id`) AS total, type, start, adjust FROM `event`
-			WHERE `event`.`uid` = %d AND `start` < '%s' AND `finish` > '%s' and `ignore` = 0
-			ORDER BY `start` ASC ",
-			intval(local_user()),
-			dbesc(datetime_convert('UTC', 'UTC', 'now + 7 days')),
-			dbesc(datetime_convert('UTC', 'UTC', 'now'))
-		);
+		$cachekey = "ping_init:".local_user();
+		$ev = Cache::get($cachekey);
```

Comment:
It is no problem to only update this value once in a while. The system tells about upcoming events up to seven days in the future. On hour less or more really doesn't matter.